### PR TITLE
Starting point for 630/It3

### PR DESCRIPTION
**Pull request by vitokhangnguyen**
This PR addresses the following matters:

* Remake `OrganizationDashboardComponent` with buttons to Location and Personnel page. These buttons are only enabled if the user has the right claim.
* Implement a sufficient-enough-for-this-iteration access control system so that only users with `isManager` claim can access the `/organization/location` and `/organization/personnel` routes. (Please ask me any question if needed regarding the claims and access control)
* Create placeholders for `LocationManagementComponent` (my part) and `PersonnelManagementComponent` (@rayshroom's part).
* Fix some harmless (but ugly) exceptions thrown in the production build.
* Add images to display when the product's photo fails to load (due to my mistake of deleting them accidentally last iteration and my laziness to add them back).

If possible, let's have this PR landed into the `develop` branch before we can start our Iteration 3.

### CODE REVIEW

File: src/app/services/organization/organization-role.guard.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    state: RouterStateSnapshot): boolean {
+    const currentOrg = this.o.getCurrentOrganization();
+    const requiredClaims = next.data.requiredClaims as Array<string>;
+    // tslint:disable-next-line: prefer-for-of
+    for (let i = 0; i < requiredClaims.length; i++) {
+        if (!currentOrg.claims[requiredClaims[i]]) {
+            this.router.navigate(['/organization']);
+            return false;
+        }
+    }
```
Comment (Silvyre) [inline src/app/services/organization/organization-role.guard.ts]: I think tslint wants this:
```js
    for (claim of requiredClaims) {
        if (!currentOrg.claims[claim]) {
            this.router.navigate(['/organization']);
            return false;
        }
    }
```

File: src/environments/environment.ts [hunk 2]
```diff
@@ -14,8 +14,8 @@ export const environment = {
         appId: '1:216500507265:web:ef43d619b4ac117655f380',
         measurementId: 'G-3FKLY60L44'
     },
-    // api: 'http://localhost:8080/api',
-    api: 'https://inventurbo-api.herokuapp.com/api',
+    api: 'http://localhost:8080/api',
+    // api: 'https://inventurbo-api.herokuapp.com/api',
```
Comment (Silvyre) [inline src/environments/environment.ts]: Necessary change? I didn't see any updates to the backend in this PR.

**Comment by Silvyre** [comment]
> Implement a sufficient-enough-for-this-iteration access control system so that only users with `isManager` claim can access the `/organization/location` and `/organization/personnel` routes. (Please ask me any question if needed regarding the claims and access control)

I added `isManager` and `isOwner` to my OrganizationLink document (I only have one); that's enough, right? Do we have a way to specify claims during account registration yet?

File: src/app/services/organization/organization-role.guard.ts (hunk 1 above)
Comment (vitokhangnguyen) [reply src/app/services/organization/organization-role.guard.ts]: You are correct, sir!

File: src/environments/environment.ts (hunk 2 above)
Comment (vitokhangnguyen) [reply src/environments/environment.ts]: Forgot to mention that I push a small change to the backend. Besides that, this change allows us to modify the backend and use it locally during the iteration. I will change it back in the end before landing everything to master.

**Comment by vitokhangnguyen** [comment]
I will elaborate more on the claim thing in a moment.

**Comment by vitokhangnguyen** [comment]
I will elaborate more on the claim thing in a moment.

**Comment by vitokhangnguyen** [comment]
So 2 things to elaborate:

* On the backend side, when a user creates an organization, they will be assigned the role of `Owner` and 2 claims of `isOwner` and `isManager`. The role is just to display on the UI and should not be used to perform access/permission controls, that is rather the purpose of claims.

* My plan is that **1 organization** should only has **1 owner** but can has **multiple managers**. When an owner makes someone a manager, they will be rewarded with **only the claim `isManager`**. The claim `isOwner` is only used in use cases such as giving and revoking manager privilege or changing organization's critical info. Otherwise, the claim `isManager` should be sufficient.

And yes, @Silvyre, your change is good neough. Although I also put you out of retired and get you back to become `Owner` role.

**Comment by vitokhangnguyen** [comment]
Updated the code. Merging now.

**Pull request merged**